pointer/pointer_test.go

```diff
@@ -372,3 +373,49 @@ func TestFloat64Equal(t *testing.T) {
 		t.Errorf("expected false (val != val)")
 	}
 }
+
+func TestDuration(t *testing.T) {
+	val := time.Duration(0)
+	ptr := Duration(val)
+	if *ptr != val {
+		t.Errorf("expected %s, got %s", val, *ptr)
+	}
+
+	val = time.Duration(42)
+	ptr = Duration(val)
+	if *ptr != val {
+		t.Errorf("expected %s, got %s", val, *ptr)
+	}
+}
+
+func TestDurationDeref(t *testing.T) {
+	var val, def time.Duration = 42, 0
+
+	out := DurationDeref(&val, def)
+	if out != val {
+		t.Errorf("expected %s, got %s", val, out)
+	}
+
+	out = DurationDeref(nil, def)
+	if out != def {
+		t.Errorf("expected %s, got %s", def, out)
+	}
+}
+
+func TestDurationEqual(t *testing.T) {
+	if !DurationEqual(nil, nil) {
+		t.Errorf("expected true (nil == nil)")
+	}
+	if !DurationEqual(Duration(42), Duration(42)) {
+		t.Errorf("expected true (val == val)")
+	}
+	if DurationEqual(nil, Duration(42)) {
+		t.Errorf("expected false (nil != val)")
+	}
+	if DurationEqual(Duration(42), nil) {
+		t.Errorf("expected false (val != nil)")
+	}
+	if DurationEqual(Duration(42), Duration(43)) {
+		t.Errorf("expected false (val != val)")
+	}
+}
```

Comment:
These look good!
Would it be possible to add these tests in a tabular form?

To be clear, I mean something like:
```go
func TestDuration(t *testing.T) {
    cases := []struct{
        name string // if applicable, optional
        val time.Duration
        ptr *time.Duration
    } {
        {...},
    }
    for _, case := range cases {
        ...
    }
}
```

Reply:
These tests are "inspired" by the existing tests that all follow the same approach. For consistency reasons, I would rather keep it as it is. I guess you can open issue to rework all existing tests to table tests. WDYT?

Reply:
Let's keep it as is :)